rust/kernel/miscdev.rs

```diff
@@ -124,15 +126,15 @@ impl<T: FileOperations> Registration<T> {
     /// Registers a miscellaneous device.
     ///
     /// Returns a pinned heap-allocated representation of the registration.
-    pub fn new_pinned(name: &'static CStr, open_data: T::OpenData) -> Result<Pin<Box<Self>>> {
+    pub fn new_pinned(name: fmt::Arguments<'_>, open_data: T::OpenData) -> Result<Pin<Box<Self>>> {
         Options::new().register_new(name, open_data)
     }
 
     /// Registers a miscellaneous device with the rest of the kernel.
     ///
     /// It must be pinned because the memory block that represents the registration is
     /// self-referential.
-    pub fn register(self: Pin<&mut Self>, name: &'static CStr, open_data: T::OpenData) -> Result {
+    pub fn register(self: Pin<&mut Self>, name: fmt::Arguments<'_>, open_data: T::OpenData) -> Result {
```

Comment:
Maybe accept `Cow<'static, CStr>` instead? That allows avoiding an allocation in the common case of the name being constant. Or `impl Into<Cow<'static, CStr>>` to allow directly passing in both a `CStr` and `CString` without having to construct a `Cow<CStr>`.

Reply:
I'd like to leave this as is now, and once #678 is done (or as part of it) we can optimise this avoid the allocation when using a `&'static CStr`